### PR TITLE
Update DPC++ extension table to point to SYCL 2020 provisional instead of extensions, where appropriate

### PR DESCRIPTION
**Pull request by mkinsner**

### CODE REVIEW

File: source/elements/dpcpp/source/index.rst [hunk 1]
```diff
@@ -48,104 +48,150 @@ some additional extensions being required, and some DPC++ extensions no longer
...
+
+-  Accessor simplifications - simplification of the accessor interface, reduction of verbosity in common
+   code, and removal of need to specify template arguments in common cases.  Section 4.7.6 of the
+   `SYCL 2020 Provisional Specification`_.
+-  bit_cast - inclusion of C++20 (p0476r2) ``std::bit_cast`` as ``sycl::bit_cast``.  Section 3.8.2 of the
+   `SYCL 2020 Provisional Specification`_.
+-  Deduction guides - simplifies common code patterns and reduces code length and verbosity by enabling
+   Class Template Argument Deduction (CTAD) from modern C++.  Distributed throughout the
+   `SYCL 2020 Provisional Specification`_.
+-  Device specific queries - kernel property queries associted with a specific device.  Section 4.12 of the
```
Comment (rscohn2) [inline source/elements/dpcpp/source/index.rst:132]: associted  -> associated

**Comment by rscohn2** [comment]
Can you add something like this? MKL is doing it, too.

Acknowledgement
---------------
We thank the DPC++ & oneDPL `Technical Advisory Board <https://github.com/oneapi-src/oneAPI-tab>`__ for their valuable feedback.

File: source/elements/dpcpp/source/index.rst (hunk 1 above)
Comment (mkinsner) [reply source/elements/dpcpp/source/index.rst]: Fixed

**Comment by mkinsner** [comment]
> Can you add something like this? MKL is doing it, too.
> 
> ## Acknowledgement
> We thank the DPC++ & oneDPL `Technical Advisory Board <https://github.com/oneapi-src/oneAPI-tab>`__ for their valuable feedback.

I've added some wording.  Please check that you're ok with it.

**Pull request merged**